### PR TITLE
Enable LTO in github actions builds

### PR DESCRIPTION
**Pull request by Rokkubro**
Enables LTO when building in the github actions. Needed some minor changes to get working:

- Windows needs a larger pagefile as otherwise the linking process runs out of memory
- Android needs to use a more recent version of cmake; otherwise it attempts to use the gold linker which is not installed
- Linux builds(excepting linux-frozen) need to use clang as it can use "ThinLTO" which has much lower memory usage and does not crash in github actions
  - To do this [changes to the build images](https://github.com/citra-emu/build-environments/pull/50) are required

Should squeak out a little bit more performance. I wasn't able to see much of a difference in the Windows or AppImage builds but I was getting a couple of frames more on my weaker android phone.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6609)

### CODE REVIEW

**Comment by FearlessTobi** [comment]
Please squash your commits into one.

**Review by FearlessTobi — CHANGES_REQUESTED** [review]
I do not agree with changing compilers, just to use LTO.

Instead we should profile which parts of the code actually benefit from using it and only enable it for them.

**Comment by Rokkubro** [comment]
> I do not agree with changing compilers, just to use LTO.
> 
> Instead we should profile which parts of the code actually benefit from using it and only enable it for them.

Alright that makes sense. In the meantime, it might be beneficial to enable LTO for the platforms which don't require a change in compiler? Or at least for android as it saw some performance benefits?

File: src/android/app/build.gradle [hunk 1]
```diff
@@ -96,7 +96,7 @@ android {
 
     externalNativeBuild {
         cmake {
-            version "3.22.1"
+            version "3.26.3+"
```
Comment (SachinVin) [inline src/android/app/build.gradle:99]: 3.22.1 is the version provided by android studio and given android's janky interaction with native code I would prefer it to be the one officially supported by android studio

**Comment by Steveice10** [comment]
Superseded by https://github.com/citra-emu/citra/issues/6887

**Pull request closed without merging**